### ISSUE TITLE
Second Gitlab CI run fails with exception

### ISSUE THREAD

**Issue by sobolevn** [danger/danger#1112]
# Report

## What did you do?

ℹ Please replace this with what you did.

I am using this setup: https://github.com/wemake-services/kira-review
The first CI commit in a PR works fine.

<img width="912" alt="Снимок экрана 2019-03-28 в 22 36 52" src="https://user-images.githubusercontent.com/4660275/55187399-00ba2480-51aa-11e9-8a52-9e5540d40c44.png">

However, when I push next commit my CI job fails.

Output:

```
$ danger --fail-on-errors=true --fail-if-no-pr=true --dangerfile=/danger/Dangerfile
/danger/vendor/bundle/ruby/2.3.0/gems/gitlab-4.10.0/lib/gitlab/request.rb:67:in `validate': Server responded with code 400, message: Bad Request. Request URI: https://gitlab.com/api/v4/projects/wemake%2Eservices%2Frostelecom_svetochka/merge_requests/214/discussions/b3c2b2195a67fcb31bce7b85bf1e4c190fe53e40/notes/155245401 (Gitlab::Error::BadRequest)
	from /danger/vendor/bundle/ruby/2.3.0/gems/gitlab-4.10.0/lib/gitlab/request.rb:46:in `block (2 levels) in <class:Request>'
	from /danger/vendor/bundle/ruby/2.3.0/gems/gitlab-4.10.0/lib/gitlab/client/merge_requests.rb:284:in `update_merge_request_discussion_note'
	from /danger/vendor/bundle/ruby/2.3.0/gems/danger-6.0.1/lib/danger/request_sources/gitlab.rb:346:in `block in submit_inline_comments!'
	from /danger/vendor/bundle/ruby/2.3.0/gems/danger-6.0.1/lib/danger/request_sources/gitlab.rb:342:in `each'
	from /danger/vendor/bundle/ruby/2.3.0/gems/danger-6.0.1/lib/danger/request_sources/gitlab.rb:342:in `submit_inline_comments!'
	from /danger/vendor/bundle/ruby/2.3.0/gems/danger-6.0.1/lib/danger/request_sources/gitlab.rb:187:in `update_pull_request_with_inline_comments!'
	from /danger/vendor/bundle/ruby/2.3.0/gems/danger-6.0.1/lib/danger/request_sources/gitlab.rb:154:in `update_pull_request!'
	from /danger/vendor/bundle/ruby/2.3.0/gems/danger-6.0.1/lib/danger/danger_core/dangerfile.rb:247:in `post_results'
	from /danger/vendor/bundle/ruby/2.3.0/gems/danger-6.0.1/lib/danger/danger_core/dangerfile.rb:278:in `run'
	from /danger/vendor/bundle/ruby/2.3.0/gems/danger-6.0.1/lib/danger/danger_core/executor.rb:29:in `run'
	from /danger/vendor/bundle/ruby/2.3.0/gems/danger-6.0.1/lib/danger/commands/runner.rb:72:in `run'
	from /danger/vendor/bundle/ruby/2.3.0/gems/claide-1.0.2/lib/claide/command.rb:334:in `run'
	from /danger/vendor/bundle/ruby/2.3.0/gems/danger-6.0.1/bin/danger:5:in `<top (required)>'
	from /danger/vendor/bundle/ruby/2.3.0/bin/danger:23:in `load'
	from /danger/vendor/bundle/ruby/2.3.0/bin/danger:23:in `<main>'
ERROR: Job failed: exit code 1
```

## What did you expect to happen?

ℹ Please replace this with what you expected to happen.

I expected `danger` to renew the comment.

## What happened instead?

ℹ Please replace this with of what happened instead.
💥

## Your Environment

* Which CI are you running on? gitlab.com
* Are you running the latest version of Danger? https://github.com/wemake-services/kira-review/blob/master/Gemfile
* What is your Dangerfile? https://github.com/wemake-services/kira-review/blob/master/Dangerfile

**Comment by sobolevn** [comment]
If I delete previous comment and rerun the same job - it passes.

```
$ danger --fail-on-errors=true --fail-if-no-pr=true --dangerfile=/danger/Dangerfile
Results:

Warnings:
- [ ] <a href='https://gitlab.com/wemake.services/rostelecom_svetochka/blob/dd321df0a10b487fadc42263623ea8a2542c3bac/.gitlab-ci.yml'>.gitlab-ci.yml</a> was edited
- [ ] <a href='https://gitlab.com/wemake.services/rostelecom_svetochka/blob/dd321df0a10b487fadc42263623ea8a2542c3bac/pyproject.toml'>pyproject.toml</a> was edited.

Messages:
- [ ] Before getting a real human to review your code I am here to help you.

  Make sure that you are familiar with our
  [merge request workflow](https://wemake.services/meta/rsdp/creating-merge-requests).

  Long story short, you will need to fix critical issues,
  otherwise your CI build will not pass.
  Warnings are also required to be fixed, but there are exceptions.

  Reviewer should ask you about unfixed warnings.

  If you have any questions:
  1. Read docs: https://github.com/wemake-services/kira-review
  2. Ping your architect, ask right away
  3. Create a bug, that something is not working / is not clear

  Have fun!
Job succeeded
```

**Comment by orta** [comment]
Maybe there are access right issues with the token for the notes?

**Comment by sobolevn** [comment]
User is added as a maintainer, access token includes `api`. What other information do you need?

**Comment by sobolevn** [comment]
I hope this will help:
1. When I click on the [link from logs](https://gitlab.com/api/v4/projects/wemake.services%2Frostelecom_svetochka/merge_requests/214/discussions/b3c2b2195a67fcb31bce7b85bf1e4c190fe53e40/notes/155245401) this is what I get: `{"message":"404 Not found"}`. I am authorised, other API methods work for me
2. I have created a public reproduction repo, but I was not able to reproduce the issue. Please take a look: https://gitlab.com/wemake.services/danger-bug1112/merge_requests/1

**Comment by sobolevn** [comment]
Ok, I cannot reproduce the issue now. I believe, that this is because of the `sticky: true` which I have removed recently. https://github.com/wemake-services/kira-review/commit/2500650815dc840a96bd9ad88f9ff327a92f2a29

I am closing this issue, thanks for your time!
